src/prpy/perception/apriltags.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2015, Carnegie Mellon University
+# All rights reserved.
+# Authors: Jennifer King <jeking@cs.cmu.edu>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# - Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+# - Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+# - Neither the name of Carnegie Mellon University nor the names of its
+#   contributors may be used to endorse or promote products derived from this
+#   software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import logging
+from base import PerceptionModule, PerceptionMethod
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+class ApriltagsModule(PerceptionModule):
+    
+    def __init__(self, marker_topic, marker_data_path, kinbody_path,
+                 detection_frame, destination_frame):
+
+        super(ApriltagsModule, self).__init__()
+        
+        self.marker_topic = marker_topic
+        self.marker_data_path = marker_data_path
+        self.kinbody_path = kinbody_path
+        self.detection_frame = detection_frame
+        self.destination_frame = destination_frame
+        
+        
+    def __str__(self):
+        return 'Apriltags'
+
+    def DetectObjects(self, robot, object_names, **kw_args):
```

Comment:
None of these functions seem to use `robot`. Instead, they always query `robot.GetEnv()`. Can we change all of these functions to take an `Environment` directly?